src/Umbraco.Core/Migrations/Upgrade/UmbracoPlan.cs

```diff
@@ -41,23 +40,33 @@ public override string InitialState
                 if (!SemVersion.TryParse(ConfigurationManager.AppSettings[Constants.AppSettings.ConfigurationStatus], out var currentVersion))
                     throw new InvalidOperationException($"Could not get current version from web.config {Constants.AppSettings.ConfigurationStatus} appSetting.");
 
-                // we currently support upgrading from 7.10.0 and later
-                if (currentVersion < new SemVersion(7, 10))
-                    throw new InvalidOperationException($"Version {currentVersion} cannot be migrated to {UmbracoVersion.SemanticVersion}.");
-
                 // cannot go back in time
                 if (currentVersion > UmbracoVersion.SemanticVersion)
                     throw new InvalidOperationException($"Version {currentVersion} cannot be downgraded to {UmbracoVersion.SemanticVersion}.");
 
-                // upgrading from version 7 => initial state is eg "{init-7.10.0}"
-                // anything else is not supported - ie if 8 and above, we should have an initial state already
-                if (currentVersion.Major != 7)
-                    throw new InvalidOperationException($"Version {currentVersion} is not supported by the migration plan.");
+                // only from 7.14.0 and above
+                var minVersion = new SemVersion(7, 14);
+                if (currentVersion < minVersion)
+                    throw new InvalidOperationException($"Version {currentVersion} cannot be migrated to {UmbracoVersion.SemanticVersion}."
+                                                        + $" Please upgrade first to at least {minVersion}.");
 
+                // initial state is eg "{init-7.14.0}"
                 return "{init-" + currentVersion + "}";
             }
         }
 
+        protected override void ThrowOnUnknownInitialState(string state)
+        {
+            if (state.StartsWith("{init-") && state.EndsWith("}"))
+            {
+                var initVersion = state.TrimStart("{init-").TrimEnd("}");
```

Comment:
Please save `"{init-"` and `"}"` in local constants, instead of having them twice as magic strings

Reply:
yes. fixing.